From 9b347a5ee9604d4ec972506328ee44a0d8eb1bf0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 14 May 2021 06:32:02 -0700
Subject: [PATCH] Makefile: use ifdefg from dev86-native instead of target

* recipe is restricted to:
  COMPATIBLE_HOST = "(i.86|x86_64).*-linux"
  so it's less likely to reproduce invalid combination of
  host and target, but I guess if you have aarch64 build
  machine and build for x86 then it might fail as for Bruce:

| ./ifdefg -MU  makefile.in >tmp.mak
| /bin/sh: 1: ./ifdefg: not found
| make[1]: *** [Makefile:46: make.fil] Error 127

  use ifdefg from native build to increase chances that it
  will be compatible with your host system

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8a61b65..7fdf861 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@ realclean:
 	-for X in */; do cd $$X && make clean; cd $(WD); done
 
 make.fil: $(IFDEFNAME) makefile.in
-	./$(IFDEFNAME) -MU $(IFDEFOPTS) makefile.in >tmp.mak
+	$(IFDEFNAME) -MU $(IFDEFOPTS) makefile.in >tmp.mak
 	echo > tmp.sed
 	[ "$(BINDIR)" != "//bin" ] || echo >> tmp.sed "s:%BINDIR%:/bin:"
 	[ "$(LIBDIR)" != "//lib/bcc" ] || echo >> tmp.sed "s:%LIBDIR%:/lib/bcc:"
